### PR TITLE
*: Add translations from spec 3.2.0

### PR DESCRIPTION
**Pull request by arithx**

### CODE REVIEW

**Comment by arithx** [comment]
I wasn't sure if there was policy behind when we update `internal/` so I left it out and just added the library functions for `v32tov2x`

File: translate/v32tov22/v32tov22.go [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+		return old.Config{}, fmt.Errorf("HTTP proxies in Ignition.Proxy are not supported on 2.2")
+	}
+
+	if cfg.Storage.Luks != nil {
+		return old.Config{}, fmt.Errorf("LUKS is not supported on 2.2")
+	}
+
+	// ShouldExist for Users & Groups do not exist in 2.2
+	for _, u := range cfg.Passwd.Users {
+		if u.ShouldExist != nil {
```
Comment (bgilbert) [inline translate/v32tov22/v32tov22.go:75]: We can also allow `*true`.

File: translate/v32tov22/v32tov22.go [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+
+	// Size and Start are sectors not MiB in 2.2, so we don't understand them.
+	// Resize is not in 2.2
+	// Fail for now
+	for _, d := range cfg.Storage.Disks {
+		for _, p := range d.Partitions {
+			if p.SizeMiB != nil || p.StartMiB != nil {
+				return old.Config{}, fmt.Errorf("SizeMiB and StartMiB in Storage.Disks.Partitions is not supported on 2.2")
+			}
+			if p.Resize != nil {
```
Comment (bgilbert) [inline translate/v32tov22/v32tov22.go:93]: We can also allow `*false`.

File: translate/v32tov22/v32tov22.go [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+		if ca.HTTPHeaders != nil {
+			return old.Config{}, fmt.Errorf("HTTPHeaders in Ignition.Security.TLS.CertificateAuthorities are not supported on 2.2")
+		}
+	}
+
+	if cfg.Ignition.Proxy.HTTPProxy != nil || cfg.Ignition.Proxy.HTTPSProxy != nil || cfg.Ignition.Proxy.NoProxy != nil {
+		return old.Config{}, fmt.Errorf("HTTP proxies in Ignition.Proxy are not supported on 2.2")
+	}
+
+	if cfg.Storage.Luks != nil {
```
Comment (bgilbert) [inline translate/v32tov22/v32tov22.go:69]:
```
if len(cfg.Storage.Luks) > 0 {
```

**Review by bgilbert — COMMENTED** [review]
I don't think we need to update `internal` for now.

**Comment by arithx** [comment]
Updated

**Review by bgilbert — APPROVED** [review]

**Pull request merged**